packages/noya-state/src/reducers/layerReducer.ts

```diff
@@ -195,6 +182,136 @@ export function layerReducer(
         updateSelection(draft.selectedObjects, id, selectionType);
       });
     }
+    case 'addSymbol': {
+      const [, id, name] = action;
```

Comment:
We should definitely figure out a way to share a lot of this logic with creating groups. Maybe we can start with a helper function in this file?

Reply:
I'll make a couple of smaller functions of the shared actions and then we can see which ones need to be together